tests/internal/test_settings.py

```diff
@@ -103,6 +103,13 @@ def _deleted_rc_config():
             "expected": {"trace_http_header_tags": {"header": "value"}},
             "expected_source": {"trace_http_header_tags": "code"},
         },
+        {
```

Comment:
Do you think it's worth having a test for the condition where someone tries to mix and match the two formats, ie:

`DD_TRACE_HEADER_TAGS="X-Header-Tag-1,X-Header-Tag-2:header_tag_2"`

to make sure that this scenario also works?

(In this case, X-Header-Tag-1 would just be left to the default logic, while X-Header-Tag-2 gets remapped as header_tag_2).